### PR TITLE
Add diagnostic and audit log v2 UI

### PR DESCRIPTION
**Pull request by sahandilshan**
### Purpose
- $subject.
- Add support to view and download audit log
- Change the Logs API endpoints to v2

https://github.com/wso2/identity-apps/assets/32576163/ce3d0a0a-04a9-40af-86ba-38466e1b1093

### CODE REVIEW

File: apps/console/src/extensions/components/audit-logs/index.ts
Comment (SujanSanjula96) [inline apps/console/src/extensions/components/audit-logs/index.ts:1]: Let's get rid of index.ts files to get rid of cyclic imports

**Comment by sahandilshan** [comment]
There might be an issue with the linter, though it suggest that there are tailing spaces, these files doesn't have any tailing spaces on the given line below
Screenshot 2023-11-22 at 7 15 53 PM

**Comment by JayaShakthi97** [comment]
@sahandilshan, IDE shows trailing spaces in above said lines.

Ex: fr-FR/extensions.ts
image

File: apps/console/src/extensions/i18n/models/extensions.ts [hunk 1]
```diff
@@ -2532,6 +2533,9 @@ export interface Extensions {
                 queryButton: {
                     label: string;
                 };
+                downloadButton : {
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/i18n/models/extensions.ts:2536]:
```suggestion
                downloadButton: {
```
Check other places as well.

File: apps/console/src/extensions/components/audit-logs/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply apps/console/src/extensions/components/audit-logs/index.ts]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/components/logs/api/logs.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpMethods } from "@wso2is/core/models";
+import useRequest, { RequestConfigInterface, RequestResultInterface } from
+    "../../../../features/core/hooks/use-request";
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply apps/console/src/extensions/components/logs/api/logs.ts]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 4]
```diff
@@ -224,43 +251,56 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
-                                                                }
-                                                            </small>
-                                                        ) }
-                                                        position="bottom right"
-                                                        size="mini"
-                                                    />
-                                                ) : log.resultMessage
-                                            }
-                                        </div>
+                                        { (logType === 0) &&
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 5]
```diff
@@ -224,43 +251,56 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                                            (<div className="log-message-container">
+                                                { log.resultStatus === "SUCCESS" &&
+                                                    <Icon name="check circle" color="green" />
+                                                }
+                                                { log.resultStatus === "FAILED" &&
+                                                    <Icon name="close" color="red" />
+                                                }
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/configs/endpoints.ts [hunk 6]
```diff
@@ -32,6 +32,7 @@ export const getExtendedFeatureResourceEndpoints = (serverHost: string,
     const authzServiceHost: string = deploymentConfig.extensions?.authzServiceHost as string;
 
     return {
+        auditLogsEndpoint:  `${ serverHost }/api/asgardeo/v2/logs/audit/search`,
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply apps/console/src/extensions/configs/endpoints.ts]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/i18n/models/extensions.ts (hunk 1 above)
Comment (sahandilshan) [reply apps/console/src/extensions/i18n/models/extensions.ts:2536]: Fixed with dd326a4fcaabec77fd74c72f6b9c3a4346203585

File: apps/console/src/extensions/components/logs/api/index.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (Achintha444) [inline apps/console/src/extensions/components/logs/api/index.ts:2]:
```suggestion
 * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/extensions/components/logs/api/logs.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
+/**
```
Comment (Achintha444) [inline apps/console/src/extensions/components/logs/api/logs.ts:1]: rename the file to `useLogs .ts`

File: apps/console/src/extensions/components/logs/api/index.ts (hunk 7 above)
Comment (Achintha444) [reply apps/console/src/extensions/components/logs/api/index.ts:2]: Check other places.

Comment (sahandilshan) [reply apps/console/src/extensions/components/logs/api/index.ts]: Fixed with f159ffb40208f9d92ad1a9fb1965e1609baf6862

File: apps/console/src/extensions/components/logs/api/logs.ts (hunk 8 above)
Comment (sahandilshan) [reply apps/console/src/extensions/components/logs/api/logs.ts]: Fixed with Fixed with f159ffb40208f9d92ad1a9fb1965e1609baf6862

File: apps/console/src/extensions/components/logs/api/index.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (Achintha444) [inline apps/console/src/extensions/components/logs/api/index.ts:2]: Shall we remove this index file

File: apps/console/src/extensions/components/logs/models/index.ts [hunk 10]
```diff
@@ -16,4 +16,4 @@
  * under the License.
  */
 
-export * from "./diagnostic-log-models";
```
Comment (Achintha444) [inline apps/console/src/extensions/components/logs/models/index.ts:19]: Shall we remove this index file

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**